File: Urbiss.Domain/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Urbiss.Domain.Models
{
    public class Order : BaseEntity
    {
        [Column(TypeName = "character varying(40)")]
        [Required]
        public string OrderId { get; set; }

        [Column]
        [Required]
        public int UserId { get; set; }

        [Column]
        [Required]
        public DateTime Date { get; set; }

        [Column]
        [Required]
        public long UserSurveyId { get; set; }
        public UserSurvey UserSurvey { get; set; }

        [Column]
        [Required]
        public long SurveyId { get; set; }

        [Column]
        [Required]
        public double Price { get; set; }

        [Column]
        public long? VoucherId { get; set; }

        [Column(TypeName = "character varying(15)")]
        [Required]
        public OrderStatusEnum Status { get; set; } = OrderStatusEnum.Created;
    }

    public enum OrderStatusEnum
    {
        Created,
        Processed
    }
}

File: app/src/app/shared/order.ts
export class Order {
  UserSurveyId: number;
  surveyid: number;
  voucherCode: string;

  public constructor(
    _UserSurveyId: number, 
	_surveyid: number,
	_voucherCode: string,
  ) {
    this.UserSurveyId = _UserSurveyId;
    this.surveyid = _surveyid;
	this.voucherCode = _voucherCode;
  }
}
  